error[E0308]: mismatched types in lib.rs:108:1

I don't know anything about rust, or this project. I am using cargo to install another binary which has tower-lsp v0.14.1 as a dependency and I'm seeing this error:

```
error[E0308]: mismatched types
   --> /home/bphunter/.cargo/registry/src/github.com-1ecc6299db9ec823/tower-lsp-0.14.1/src/lib.rs:108:1
    |
108 | #[rpc]
    | ^^^^^^ expected enum `std::option::Option`, found enum `Id`
    |
    = note: expected enum `std::option::Option<Id>`
               found enum `Id`
    = note: this error originates in the attribute macro `rpc` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try wrapping the expression in `jsonrpc::_::_serde::__private::Some`
    |
108 | jsonrpc::_::_serde::__private::Some(#[rpc])
    | ++++++++++++++++++++++++++++++++++++      +

For more information about this error, try `rustc --explain E0308`.
error: could not compile `tower-lsp` due to previous error
```

I am using: cargo 1.58.0 (f01b232bc 2022-01-19)

Any help or explanation would be appreciated.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,7 +7,15 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
-## [0.15.0] - 2022-02-10
+## [0.15.1] - 2022-02-14
+
+### Fixed
+
+* Fix semver incompatibility in release of `tower-lsp-macros` (PR #306).
+  * Re-released `tower-lsp-macros` 0.4.2 -> 0.5.0.
+  * Re-released `tower-lsp` 0.15.0 -> 0.15.1.
+
+## [0.15.0] - 2022-02-10 [YANKED]
 
 ### Changed
 
@@ -435,7 +443,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   * `textDocument/hover`
   * `textDocument/documentHighlight`
 
-[Unreleased]: https://github.com/ebkalderon/tower-lsp/compare/v0.15.0...HEAD
+[Unreleased]: https://github.com/ebkalderon/tower-lsp/compare/v0.15.1...HEAD
+[0.15.1]: https://github.com/ebkalderon/tower-lsp/compare/v0.15.0...v0.15.1
 [0.15.0]: https://github.com/ebkalderon/tower-lsp/compare/v0.14.1...v0.15.0
 [0.14.1]: https://github.com/ebkalderon/tower-lsp/compare/v0.14.0...v0.14.1
 [0.14.0]: https://github.com/ebkalderon/tower-lsp/compare/v0.13.3...v0.14.0

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "tower-lsp"
-version = "0.15.0"
+version = "0.15.1"
 authors = ["Eyal Kalderon <ebkalderon@gmail.com>"]
 edition = "2018"
 description = "Language Server Protocol implementation based on Tower"
@@ -26,7 +26,7 @@ serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 tokio = "1.6"
 tokio-util = { version = "0.7.0", features = ["codec"] }
-tower-lsp-macros = { version = "0.4.2", path = "./tower-lsp-macros" }
+tower-lsp-macros = { version = "0.5.0", path = "./tower-lsp-macros" }
 tower-service = "0.3"
 
 [dev-dependencies]

--- a/tower-lsp-macros/Cargo.toml
+++ b/tower-lsp-macros/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "tower-lsp-macros"
-version = "0.4.2"
+version = "0.5.0"
 authors = ["Eyal Kalderon <ebkalderon@gmail.com>"]
 description = "Internal procedural macros for tower-lsp"
 edition = "2018"

